content/general/push/activate-subscribe.textile

```diff
@@ -44,31 +38,39 @@ Before you can activate your push device or receive push notifications, you must
+If you do not have a firebase project, create a new firebase project and add your Android app to the project. Then setup "firebase cloud messaging":https://firebase.google.com/docs/cloud-messaging/android/client in your Android application. Then "create and add a server key":https://knowledge.ably.com/where-can-i-find-my-google/firebase-cloud-messaging-api-key on your firebase project, and enter it in your Ably app's dashboard (App > Notifications tab > Push Notifications Setup > Setup Push Notifications). 
 
-The second requirement is that your app integrates with Firebase to handle registration token notifications, and these notifications do need to be passed to Ably; the registration token is used by Ably's servers to be able to push messages to each specific device. You can be notified of token updates either by overriding "@onNewToken()@":https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessagingService in your derived "@FirebaseMessagingService@":https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessagingService, or by overriding "@onTokenRefresh()@":https://firebase.google.com/docs/reference/android/com/google/firebase/iid/FirebaseInstanceIdService in a service that derives from "@FirebaseInstanceIdService@":https://firebase.google.com/docs/reference/android/com/google/firebase/iid/FirebaseInstanceIdService.
+To receive messages on an Android device, you need to implement a service that extends "@FirebaseMessagingService@":https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessagingService. To handle incoming push notifications, override "@onMessageReceived()@":https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessagingService. Though these messages are sent through Ably, when received on the device, they (@RemoteMessage@) do not interact with the Ably clients, and you should implement logic to handle the push notification messages appropriately, such as showing a UI "notification":https://developer.android.com/guide/topics/ui/notifiers/notifications.
 
-Registration tokens must be notified to the Ably library by calling @AblyFirebaseInstanceIdService.onNewRegistrationToken()@. If deriving from "@FirebaseInstanceIdService@":https://firebase.google.com/docs/reference/android/com/google/firebase/iid/FirebaseInstanceIdService directly, for example, your class should end up looking more or less like this:
+To push messages to specific devices, Ably uses the device's registration token. Keep Ably updated with the latest device registration token by overriding "@FirebaseMessagingService#onNewToken@":https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessagingService#public-void-onnewtoken-string-token and calling "@ActivationContext#onNewRegistrationToken@":. Your service extending "FirebaseMessagingService" should look like the following:
```

Comment:
```suggestion
To push messages to a specific device, Ably uses the device's registration token. Keep Ably updated with the latest device registration token by overriding "@FirebaseMessagingService#onNewToken@":https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessagingService#public-void-onnewtoken-string-token and calling "@ActivationContext#onNewRegistrationToken@":. Your service extending "FirebaseMessagingService" should look like the following:
```

Reply:
Does it make more sense to say: 
> To push messages to specific devices, Ably uses the devices' registration tokens.

I'm wondering because my intention is to be plural instead of singular.

Reply:
Sounds good.